install.sh

```diff
@@ -0,0 +1,302 @@
+        echo
+        echo "Options:"
+        echo "  -v, --version    Specify a version (default: latest)"
+        echo "  -b, --bin-dir    Specify the installation directory (default: \$HOME/.local/bin)"
+        echo "  -t, --tmp-dir    Specify the temporary directory (default: system temp directory)"
+        echo "  -h, --help       Show this help message"
+        echo
+        echo "Note: For versions v0.4.0 and earlier, please use the installer attached to the GitHub release."
+        exit 0
+        ;;
+      *)
+        fail "Unknown option: $1"
+        ;;
+    esac
+  done
+}
+
+# Get the latest version number from GitHub
+get_latest_version() {
+  local url="https://api.github.com/repos/fossas/circe/releases/latest"
+  local version
+  
+  if ! version=$(curl -sSfL "$url" | grep -o '"tag_name": "v[^"]*"' | cut -d'"' -f4); then
+    fail "Failed to get latest version from GitHub"
+  fi
+  
+  echo "$version"
+}
+
+# Download a file
+download() {
+  local url="$1"
+  local dest="$2"
+  
+  info "Downloading $url to $dest"
+  
+  if ! curl -sSfL "$url" -o "$dest"; then
+    fail "Failed to download from $url"
+  fi
+}
+
+# Install the binary
+install_binary() {
+  local platform="$1"
+  local version="$2"
+  local bin_dir="$3"
+  local tmp_dir="$4"
+  local download_url
+  local checksums_url
+  local archive_name
+  local binary_name="circe"
+  local release_prefix="https://github.com/fossas/circe/releases/download/${version}"
+
+  # Determine archive name based on platform
+  if [[ "$platform" == *"-windows-"* ]]; then
+    archive_name="circe-${platform}.zip"
+    binary_name="circe.exe"
+  else
+    archive_name="circe-${platform}.tar.gz"
+  fi
+
+  # Construct download URLs
+  download_url="${release_prefix}/${archive_name}"
+  checksums_url="${release_prefix}/checksums.txt"
+
+  # Create temporary directory
+  local workdir="$tmp_dir/circe-install-$$"
+  mkdir -p "$workdir"
+  cd "$workdir"
+
+  # Download archive and checksums
+  download "$download_url" "$archive_name"
+  download "$checksums_url" "checksums.txt"
+
+  # Verify checksum
+  info "Verifying checksum"
+  local expected_checksum
+  expected_checksum=$(grep "$archive_name" checksums.txt | awk '{print $1}')
+  if [[ -z "$expected_checksum" ]]; then
+    fail "Couldn't find checksum for $archive_name"
+  fi
+
+  local actual_checksum
+  if command -v sha256sum > /dev/null; then
+    actual_checksum=$(sha256sum "$archive_name" | awk '{print $1}')
+  elif command -v shasum > /dev/null; then
+    actual_checksum=$(shasum -a 256 "$archive_name" | awk '{print $1}')
+  else
+    fail "Neither sha256sum nor shasum found, cannot verify download"
+  fi
+
+  if [[ "$expected_checksum" != "$actual_checksum" ]]; then
+    fail "Checksum verification failed! Expected: $expected_checksum, got: $actual_checksum"
+  fi
+
+  # Extract archive
+  if [[ "$archive_name" == *.tar.gz ]]; then
+    tar -xzf "$archive_name"
```

Comment:
I learned recently that most tars (not busybox apparently) will automatically detect the compression type if you just do it like `tar -xf`.